### PR TITLE
[Fix] end_level is the index of the end input backbone level

### PR DESCRIPTION
**Pull request by Czm369**
Thanks for your contribution and we appreciate it a lot. The following instructions would make your pull request more healthy and more easily get feedback. If you do not understand some items, don't worry, just make the pull request and seek help from maintainers.

## Motivation

end_level is the index of the end input backbone level, so `end_level=-1` is equal to `end_level=self.num_ins  - 1`

## Modification

To fix the bug, just change the if condition

To fix the bug, just change the if condition

To fix the bug,  we need to change the if condition
`if end_level == -1 or end_level == self.num_ins - 1`
and add 1 for some variables.

## BC-breaking (Optional)

Does the modification introduce changes that break the backward-compatibility of the downstream repos?
If so, please describe how it breaks the compatibility and how the downstream projects should modify their code to keep compatibility with this PR.

## Use cases (Optional)

If this PR introduces a new feature, it is better to list some use cases here, and update the documentation.

## Checklist

1. Pre-commit or other linting tools are used to fix the potential lint issues.
2. The modification is covered by complete unit tests. If not, please add more unit test to ensure the correctness.
3. If the modification has potential influence on downstream projects, this PR should be tested with downstream projects, like MMDet or MMCls.
4. The documentation has been modified accordingly, like docstring or example tutorials.

### CODE REVIEW

File: mmdet/models/necks/fpg.py [hunk 1]
```diff
@@ -212,14 +212,14 @@ def __init__(self,
...
+        if end_level == -1 or end_level == self.num_ins - 1:
             self.backbone_end_level = self.num_ins
             assert num_outs >= self.num_ins - start_level
         else:
-            # if end_level < inputs, no extra level is allowed
-            self.backbone_end_level = end_level
-            assert end_level <= len(in_channels)
-            assert num_outs == end_level - start_level
+            # if end_level is not the last level, no extra level is allowed
+            self.backbone_end_level = end_level + 1
```
Comment (ZwwWayne) [inline mmdet/models/necks/fpg.py:220]: explain why +1 is necessary here.

File: tests/test_models/test_necks.py [hunk 2]
```diff
@@ -13,6 +13,18 @@ def test_fpn():
...
+    FPN(in_channels=in_channels,
+        out_channels=out_channels,
+        start_level=0,
+        end_level=-1,
+        num_outs=5)
+    FPN(in_channels=in_channels,
+        out_channels=out_channels,
+        start_level=0,
+        end_level=3,
+        num_outs=5)
```
Comment (ZwwWayne) [inline tests/test_models/test_necks.py:27]: It seems that the updates are not fully covered by unit tests. Please add more unit tests to fix the issues.

**Review by zytx121 — APPROVED** [review]
LGTM

**Comment by hhaAndroid** [comment]
@Czm369 ci failed

**Review by hhaAndroid — COMMENTED** [review]
LGTM

**Comment by ZwwWayne** [comment]
We should not only test assertion errors, but also test the cases when they can run normally.

**Pull request merged**